sklearnex/covariance/incremental_covariance.py

```diff
@@ -0,0 +1,132 @@
+# ===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import numpy as np
+from sklearn.utils import check_array, gen_batches
+
+from daal4py.sklearn._utils import control_n_jobs, run_with_n_jobs
+from onedal._device_offload import support_usm_ndarray
+from onedal.covariance import (
+    IncrementalEmpiricalCovariance as onedal_IncrementalEmpiricalCovariance,
+)
+
+
+@control_n_jobs
+class IncrementalEmpiricalCovariance:
+    """
+    Incremental estimator for covariance.
+    Allows to compute empirical covariance estimated by maximum
+    likelihood method if data are splitted into batches.
+
+    Parameters
+    ----------
+    batch_size : int, default=None
+        The number of samples to use for each batch. Only used when calling
+        ``fit``. If ``batch_size`` is ``None``, then ``batch_size``
+        is inferred from the data and set to ``5 * n_features``, to provide a
```

Comment:
if I correctly understood, we split input data into 5 batches by default, but whats the reason of it? May be its better to avoid splitting, due to have a clear understanding that we can provide blocks with significantly different sizes

Reply:
I got your point, so, looks like we have to add it, but my suggestion is to set it equals to 1 by default in intelex. 
Correct me, if I am wrong, but it seems like:
1)we get batch from non-onedal array/table
2)convert it to onedal type
3)call partial compute
4)convert computed batch in non-onedal array/table
5)take the second batch
and in my understanding it can significantly influence on the performance, and if you can its better to measure both options and chose the best one

Reply:
The usage is the following: if the user has already split data then partial_fit method should be used. if user has the whole data then it is possible to use fit method which automatically splits data and then calls partial_fit for each batch. I updated the example to highlight it.